contrib/mesos/docs/ha.md

```diff
@@ -1,4 +1,25 @@
-## High Availability
+# High Availability
+
+Kubernetes on Mesos supports 2 HA modes:
+
+* [Hot-standby](#hot-standby)
+* [Cold-standby](#cold-standby)
+
+In the hot-standby mode, all master components (apiserver, controller manager, and scheduler) actively run 
+on every master node. Additional logic is added to the controller manager and scheduler to coordinate
+their access to the etcd backend to deal with concurrency issues when modifying cluster state. As 
+apiserver does not modify cluster state, multiple of these can run concurrently without coordination. When
+the leader (i.e., the node whose scheduler is active) crashes, other master nodes will detect
+the failure after some time and then elect a new leader. (*HA for controller manager is currently work-in-progress, 
```

Comment:
Nice that we have this footnote here but I'd rather see the paragraph re-worded to better fit.